addons/html_editor/static/src/main/media/image_plugin.js

```diff
@@ -294,7 +294,11 @@ export class ImagePlugin extends Plugin {
         }
     }
 
-    onSelectionChange() {
+    onSelectionChange(selectionData) {
+        const { anchorNode, focusNode } = selectionData.documentSelection;
+        if (!anchorNode && !focusNode) {
+            return;
+        }
```

Comment:
I don't understand why this change is needed?

Reply:
I made the following changes because removing all ranges of selection triggered the onSelectionChange event after the TRANSFORM_IMAGE action was dispatched, which caused the image transformation to close(https://github.com/odoo/odoo/blob/18.0/addons/html_editor/static/src/main/media/image_plugin.js#L294). 

Also after a rebase my fix was not working due to a recent commit (https://github.com/odoo/odoo/commit/62be82dd2c6660ef075073d61b560dfd60220c93), the behavior has changed: after dispatching the action, if the selection is no longer within the editable area, it sets the selection back into the editable. As a result, anchorNode and focusNode are no longer null, and in onSelectionChange, this causes the image transformation to close. 

One possible fix is to remove the range in the next tick.